packages/bot-web-ui/src/components/dashboard/quick-strategy/quick-strategy-components/fields.tsx

```diff
@@ -0,0 +1,228 @@
+    TDropdowns,
+    TSelectedValuesSelect,
+} from '../quick-strategy.types';
+import { TQuickStrategyFields } from './components.types';
+
+const QuickStrategyFields = React.memo(
+    ({
+        is_mobile,
+        types_strategies_dropdown,
+        symbol_dropdown,
+        trade_type_dropdown,
+        duration_unit_dropdown,
+        selected_type_strategy,
+        selected_trade_type,
+        selected_symbol,
+        selected_duration_unit,
+        onChangeDropdownItem,
+        onHideDropdownList,
+        setFieldValue,
+        onScrollStopDropdownList,
+        handleChange,
+        onChangeInputValue,
+        setCurrentFocus,
+        values,
+        description,
+        errors,
+    }: TQuickStrategyFields) => {
+        const is_double_idx_ref: React.MutableRefObject<number | undefined> = React.useRef();
+
+        const uniq_selected_input = React.useMemo(
+            () => data_uniq_input_obj.filter(elem => elem.index === selected_type_strategy.index)[0],
+            [selected_type_strategy]
+        );
+
+        const fields = React.useMemo(
+            () =>
+                data_fields.map((item, idx) => {
+                    const {
+                        id,
+                        field_name,
+                        className,
+                        label_className,
+                        field_className,
+                        input_value,
+                        select_value,
+                        label,
+                        placeholder,
+                        trailing_icon_message,
+                        zIndex,
+                        is_able_disabled,
+                        is_basic_field,
+                    } = item as TDataFields;
+
+                    const is_uniq_strategy_field = Array.isArray(item);
+                    const is_input_field = is_uniq_strategy_field || input_value;
+
+                    //dropdowns
+                    const types_strategies_options = id === 'types-strategies' && types_strategies_dropdown;
+                    const symbol_options = id === 'symbol' && symbol_dropdown;
+                    const trade_type_options = id === 'trade-type' && trade_type_dropdown;
+                    const duration_unit_options = id === 'duration-unit' && duration_unit_dropdown;
+
+                    const dropdowns: TDropdowns =
+                        types_strategies_options || symbol_options || trade_type_options || duration_unit_options || [];
+                    const dropdown_list: TDropdowns = !is_uniq_strategy_field ? dropdowns : [];
+
+                    //selected_items
+                    const type_strategy_pick = id === 'types-strategies' && selected_type_strategy;
+                    const symbol_pick = id === 'symbol' && selected_symbol;
+                    const trade_type_pick = id === 'trade-type' && selected_trade_type;
+                    const duration_unit_pick = id === 'duration-unit' && selected_duration_unit;
```

Comment:
Nit but can we create constants for all the strings in this file?

Reply:
done ✅